lib/src/main/java/com/dynatrace/file/util/PollBasedFilePoller.java

```diff
@@ -16,38 +16,30 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
 import java.time.Duration;
-import java.util.Arrays;
 import java.util.concurrent.*;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.logging.Logger;
+import java.util.zip.CRC32;
 
 class PollBasedFilePoller extends FilePoller {
   private static final Logger LOGGER = Logger.getLogger(PollBasedFilePoller.class.getName());
   private static final String LOG_MESSAGE_FAILED_FILE_READ = "Failed to read file %s. Error: %s";
 
   private final AtomicBoolean changedSinceLastInquiry = new AtomicBoolean(false);
-  private byte[] prevChecksumBytes;
 
   private final ScheduledFuture<?> worker;
   private final ScheduledExecutorService executorService;
 
-  private MessageDigest md5;
+  private final CRC32 crc32 = new CRC32();
+  private long prevChecksumValue = 0;
```

Comment:
AFAIK 0 is a valid result for CRC32 - not sure how likely this is, or if it's worth protecting against. 🤔

Reply:
In fact, it seems to be the value you get when passing in an empty byte array. The calculation of the CRC starts with the internal CRC value at 0 and is bitwise and-ed with 0xffffff: https://github.com/adoptium/jdk/blob/2c41f5adbfcebb057c2ffc8396729bdd1c100079/src/java.base/share/classes/java/util/zip/CRC32.java#L122-L133. The updateBytes function seems to call a native library (maybe zlib?), but I think its rather unlikely that it will return 0 for a non-zero input.